google-cloud-clients/google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1beta1/it/ITBigQueryStorageTest.java

```diff
@@ -373,39 +375,157 @@ public void testReadAtSnapshot() throws InterruptedException, IOException {
             .build();
 
     Job firstJob =
-        RunQueryJobAndExpectSuccess(
+        RunQueryAppendJobAndExpectSuccess(
             /* destinationTableId = */ testTableId, /* query = */ "SELECT 1 AS col");
 
     Job secondJob =
-        RunQueryJobAndExpectSuccess(
+        RunQueryAppendJobAndExpectSuccess(
             /* destinationTableId = */ testTableId, /* query = */ "SELECT 2 AS col");
 
     final List<Long> rowsAfterFirstSnapshot = new ArrayList<>();
     ProcessRowsAtSnapshot(
-        tableReference,
-        firstJob.getStatistics().getEndTime(),
-        new AvroRowConsumer() {
+        /* tableReference = */ tableReference,
+        /* snapshotInMillis = */ firstJob.getStatistics().getEndTime(),
+        /* filter = */ null,
+        /* consumer = */ new AvroRowConsumer() {
           @Override
-          public void accept(GenericRecord record) {
+          public void accept(GenericData.Record record) {
             rowsAfterFirstSnapshot.add((Long) record.get("col"));
           }
         });
     assertEquals(Arrays.asList(1L), rowsAfterFirstSnapshot);
 
     final List<Long> rowsAfterSecondSnapshot = new ArrayList<>();
     ProcessRowsAtSnapshot(
-        tableReference,
-        secondJob.getStatistics().getEndTime(),
-        new AvroRowConsumer() {
+        /* tableReference = */ tableReference,
+        /* snapshotInMillis = */ secondJob.getStatistics().getEndTime(),
+        /* filter = */ null,
+        /* consumer = */ new AvroRowConsumer() {
           @Override
-          public void accept(GenericRecord record) {
+          public void accept(GenericData.Record record) {
             rowsAfterSecondSnapshot.add((Long) record.get("col"));
           }
         });
     Collections.sort(rowsAfterSecondSnapshot);
     assertEquals(Arrays.asList(1L, 2L), rowsAfterSecondSnapshot);
   }
 
+  @Test
+  public void testColumnPartitionedTableByDateField() throws InterruptedException, IOException {
+    Field intFieldSchema =
+        Field.newBuilder("num_field", LegacySQLTypeName.INTEGER)
+            .setMode(Mode.REQUIRED)
+            .setDescription("IntegerDescription")
+            .build();
+    Field dateFieldSchema =
+        Field.newBuilder("date_field", LegacySQLTypeName.DATE)
+            .setMode(Mode.REQUIRED)
+            .setDescription("DateDescription")
+            .build();
+    com.google.cloud.bigquery.Schema tableSchema =
+        com.google.cloud.bigquery.Schema.of(intFieldSchema, dateFieldSchema);
+
+    TableId testTableId =
+        TableId.of(/* dataset = */ DATASET, /* table = */ "test_column_partition_temp_data");
+    bigquery.create(TableInfo.of(testTableId, StandardTableDefinition.of(tableSchema)));
+
+    RunQueryAppendJobAndExpectSuccess(
```

Comment:
It doesn't look like you're doing anything with the incremental jobs, so why not just do everything at once with a single CREATE TABLE AS SELECT... DDL query job here and in the other test?  It also avoids a lot of builder spam.

Reply:
Change applied in a separate commit: https://github.com/googleapis/google-cloud-java/pull/5439/commits/693f9be08244dd8cd8aa9c22f02378e1e058c8ce

The testIngestionTimePartitionedTable() could not be changed to use one DDL statement due to:
* Needs a table to be created with ingestion time partitioning
* Needs one row insert into two different partitions by using $YYYYMMDD as separate statements.

Left the test as is.